prov/rxm/src/rxm_cq.c

```diff
@@ -409,24 +409,23 @@ ssize_t rxm_cq_handle_data(struct rxm_rx_buf *rx_buf)
 
 static ssize_t rxm_handle_recv_comp(struct rxm_rx_buf *rx_buf)
 {
-	struct rxm_recv_match_attr match_attr;
+	struct rxm_recv_match_attr match_attr = {
+		.addr = FI_ADDR_UNSPEC,
+	};
 	struct dlist_entry *entry;
-	struct util_cq *util_cq;
-
-	util_cq = rx_buf->ep->util_ep.rx_cq;
 
-	if ((rx_buf->ep->rxm_info->caps & (FI_SOURCE | FI_DIRECTED_RECV)) &&
-	    !rx_buf->conn) {
-		rx_buf->conn = rxm_key2conn(rx_buf->ep, rx_buf->pkt.ctrl_hdr.conn_id);
+	if (rx_buf->ep->rxm_info->caps & FI_SOURCE) {
```

Comment:
This should be `if (rx_buf->ep->rxm_info->caps & (FI_SOURCE | FI_DIRECTED_RECV)) {` to ensure match_attr.addr is set for directed recvs.

Reply:
We have the check `if (rx_buf->ep->rxm_info->caps & FI_DIRECTED_RECV)` below that ensures match_attr.addr is set for directed recvs. This looks as follows:
```
if (rx_buf->ep->rxm_info->caps & FI_SOURCE) {
    ...
    if (rx_buf->ep->rxm_info->caps & FI_DIRECTED_RECV)
        match_attr.addr = rx_buf->conn->handle.fi_addr;
}
```

Reply:
Oh, yes. I missed this. Thanks!
Fixed it.
Now it should be okay :)